### ISSUE TITLE
whisper archive

### ISSUE THREAD

**Issue by rpk113** [graphite-project/whisper#231]
Is there a way to extract whisper data over a certain time series and place it in an archived whisper file?  This would allow data comparison at certain granularities before data aggregation occurs.

**Comment by deniszh** [comment]
Hello @rpk113,
Sorry, not really understand your question. You can extract data from whisper file (through `whisper-dump` or programmatically), of course - but if you doing that not from top archive (Archive 0) - it will be already aggregated.

**Comment by rpk113** [comment]
here is an example of what I'm trying to do..
I have CPU.wsp for my server.  I collect 1 second data from collectd and don't rollup the data for 14 days.  

I have an issue one day that is unique, but good for comparison.  so I'd like to pull that data out to use in graphs for the future.
i.e. CPU_archive_date.wsp which would have the a days worth of data in it to compare to the current.

**Comment by deniszh** [comment]
Why do you need separate archive then? Just change retention in current file and store as much unaggregated data as you want there.

**Comment by rpk113** [comment]
The issue really is file size.  If I want to compare 1 sec granularity between now and a date 4 months ago.  I would need to store 4 months of one second data which would make huge files.

**Comment by deniszh** [comment]
You will get the same size in your archived whisper file too.

**Comment by rpk113** [comment]
so really the best way to do that is to export it to CSV and plot it outside of graphite.

**Comment by deniszh** [comment]
TBH I still not really understand a problem, so, I barely able to propose a solution.
Do you need to do that for 1 metric for single time? Any solution will work.
Do you need to do that for many metrics? Then you need to "bite the bullet" and spend disk space, or switch to something else, more space optimized (whisper spending 12 bytes per point).

**Comment by rpk113** [comment]
hmm.. is it possible to disable data aggregation for certain dates?  as opposed to length of time?

**Comment by deniszh** [comment]
No, aggregation is always there, if you have more than one archive.
But if you need to storing e.g. 4 weeks with 1 second resolution- what
options do you have?

**Comment by piotr1212** [comment]
Recently, support for compressed data is added to go-carbon/go-whisper. You might want to check that out.
